eth/api_tracer.go

```diff
@@ -585,9 +586,30 @@ func (api *PrivateDebugAPI) standardTraceBlockToFile(ctx context.Context, block
 
 	// Execute transaction, either tracing all or just the requested one
 	var (
-		signer = types.MakeSigner(api.eth.blockchain.Config(), block.Number())
-		dumps  []string
+		signer      = types.MakeSigner(api.eth.blockchain.Config(), block.Number())
+		dumps       []string
+		chainConfig = api.eth.blockchain.Config()
+		canon       = true
 	)
+	// Check if there are any overrides: the caller may wish to enable a future
+	// fork when executing this block. Note, such overrides are only applicable to the
+	// actual specified block, not any preceding blocks that we have to go through
+	// in order to obtain the state.
+	// Therefore, it's perfectly valid to specify `"futureForkBlock": 0`, to enable `futureFork`
+
+	if config != nil && config.Overrides != nil {
+		// Copy the config, to not screw up the main config
+		// Note: the Clique-part is _not_ deep copied
+		chainConfigCopy := new(ctypes.ChainConfigurator)
+		*chainConfigCopy = chainConfig
+		chainConfig = *chainConfigCopy
```

Comment:
This... seems... wonky...

Reply:
@ziogaschr Check out https://github.com/etclabscore/core-geth/pull/230/commits/5a065db739295f281d478d4f38c529c7c4187482.

Reply:
I will @meowsbits. Nice use of reflect and as always proved with tests.